src/manifolds/EmbeddedManifold.jl

```diff
@@ -0,0 +1,450 @@
+"""
+    AbstractEmbeddingType
+
+A type to provide properties of the embedding an [`AbstractEmbeddedManifold`](@ref) has.
+"""
+abstract type AbstractEmbeddingType end
+
+"""
+    AbstractEmbeddedManifold{T<:AbstractEmbeddingType} <: AbstractDecoratorManifold
+
+An abstract type for an EmbeddedManifold, which acts as an [`AbstractDecoratorManifold`](@ref).
+The functions of the manifold that is embedded can hence be just passed on to the embedding.
+The embedding is further specified by an [`AbstractEmbeddingType`](@ref).
+
+This means, that formally an embedded manifold is a decorator for the manifold it is embedded
```

Comment:
Is this still true?

Reply:
Yes, but I'd rephrase that, that formally/technically the embedded manifold is a decorator of the embedding.